plugins/serverless/src/tasks/provision.ts

```diff
@@ -0,0 +1,60 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { hookFork, styles } from '@dotcom-tool-kit/logger'
+import { Task } from '@dotcom-tool-kit/types'
+import { ServerlessSchema } from '@dotcom-tool-kit/types/lib/schema/serverless'
+import { VaultEnvVars } from '@dotcom-tool-kit/vault'
+import { spawn } from 'child_process'
+
+export default class ServerlessProvision extends Task<typeof ServerlessSchema> {
+  static description = 'Provisions a job on AWS'
+
+  static defaultOptions = {
+    useVault: true,
+    buildNumVariable: 'CIRCLE_BUILD_NUM'
+  }
+
+  async run(): Promise<void> {
+    const { useVault, configPath } = this.options
+    const buildNum = process.env[this.options.buildNumVariable!]
```

Comment:
**question:** what is the `!` for?

Reply:
I think it solves this TS error:

<img width="628" alt="image" src="https://user-images.githubusercontent.com/17846996/221141028-da7e7973-3d6f-4243-b861-2c86e65db81a.png">

Without the `!` TS evaluates the type as `string | undefined`; adding the `!` asserts that the value is never null or undefined and so TS accepts it.

Reply:
...although the next line of code is 
```
if (buildNum === undefined) {
```
sooooo it _can_ be undefined even though we're asserting to TS that it won't be?

Reply:
Ahh I see! Thank you 🙏 

It looks like that `process.env[this.options.buildNumVariable]` can be undefined but we are saying the option itself cannot. And it looks like we make sure it is not undefined by defining a `defaultOption` of it above but TS hasn't made that connection.

Reply:
Yeah exactly, we're adding the default options [here](https://github.com/Financial-Times/dotcom-tool-kit/blob/18e4a271c43711a1160faed4662f41b4c5329880/lib/types/src/index.ts#L159) but TypeScript doesn't know this (we could probably solve that with an extra type parameter but it would get quite verbose). zod solves this issue 😄

Reply:
👏 great work team! 😁